File: Colosus.Client.Angular/src/models/RequestResult.ts
export class RequestResult {
  OperationName: string;
  RequestToken?: string;
  Description: string;
  Data: string;
  CreateDate: Date;
  Result: ResultEnum;

  constructor(OperationName: string) {
    this.OperationName = OperationName;
    this.CreateDate = new Date();
    this.Description = '';
    this.Data = '';
    this.Result = ResultEnum.Not;
  }

  toObject(): object {
    return {
      OperationName: this.OperationName,
      RequestToken: this.RequestToken,
      Description: this.Description,
      CreateDate: this.CreateDate,
      Result: this.Result,
    };
  }
}
export enum ResultEnum {
  Ok,
  Not,
  Stoped,
  Error,
}

File: Colosus.Entity/Concretes/RequestModel/RequestResult.cs
using Colosus.Entity.Abstracts;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Colosus.Entity.Concretes.RequestModel
{
    public class RequestResult<T> : IRequestResult
    {
        public RequestResult(string operationName)
        {
            CreateDate = DateTime.Now;
            Result = EnumRequestResult.Not;
            Description = "";
            OperationName = operationName;
        }
        public T Data { get; set; }
        public string OperationName { get; set; }
        public DateTime CreateDate { get; set; }
        public string? RequestToken { get; set; }
        public EnumRequestResult Result { get; set; }
        public string Description { get; set; }
        public string Address { get; set; }

        public bool Ok() => Result == EnumRequestResult.Ok ? true : false;
    }

    public class RequestResult : IRequestResult
    {
        public RequestResult(string operationName)
        {
            CreateDate = DateTime.Now;
            Result = EnumRequestResult.Not;
            Description = "";
            OperationName = operationName;
        }
        public string OperationName { get; set; }
        public DateTime CreateDate { get; set; }
        public string? RequestToken { get; set; }
        public EnumRequestResult Result { get; set; }
        public string Description { get; set; }
        public string Address { get; set; }

        public bool Ok() => Result == EnumRequestResult.Ok ? true : false;
    }

    public enum EnumRequestResult
    {
        Ok,
        Not,
        Stoped,
        Error,
    }
}
